### PR TITLE
Initail build of duplex steam 

### PR DESCRIPTION
**Pull request by peternewell**
Initial build of duplex steam - addition of control commands only

https://blueprints.launchpad.net/or/+spec/duplex-steam-locomotives

It is desired to merge this so that control keys are "reserved".

Additional functionality supporting this will be added later under a separate PR

### CODE REVIEW

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs
Comment (cesarBLG) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs:826]:
```suggestion
                case "engine(enginecontrollers(ortssteambooster": SteamBoosterController.Parse(stf); SteamBoosterControllerFitted = true; break;
```
I think we need an ORTS prefix there

Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs]: Thanks for pointing that out. I have now added the ORTS prefix.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs
Comment (cesarBLG) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs:7204]:
```suggestion
                Simulator.Confirmer.ConfirmWithPerCent(CabControl.SteamBooster, CabSetting.Increase, SteamBoosterController.CurrentValue * 100);
```

File: Source/Orts.Simulation/Common/Commands.cs
Comment (cesarBLG) [inline Source/Orts.Simulation/Common/Commands.cs:1594]:
```suggestion
            Receiver?.SteamBoosterChangeTo(ToState, Target);
```

**Review by cesarBLG — CHANGES_REQUESTED** [review]
Just two style fixes, sorry for not having spotted them before

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs]: Thanks, I have made this adjustment, and also to a number of other similar entries.

File: Source/Orts.Simulation/Common/Commands.cs
Comment (cesarBLG) [reply Source/Orts.Simulation/Common/Commands.cs:1594]: I'm sorry but I won't approve it as is. It looks confusing to me. In particular, at first glance your code seems to do
```csharp
if (Receiver == null)
{
    Receiver.SteamBoosterChangeTo(ToState, Target);
}
```
until you notice the "return" statement. The curly braces are completely misleading.

The ?. operator is my preferred choice in this case, and is also recommended by ORMT (see http://www.elvastower.com/forums/index.php?/topic/35509-cs-null-conditional-operator/). It is also used elsewhere in Commands.cs, so your argument about consistency is not really valid. In my opinion, new code should use it whenever possible.

However, my intention is not to force you to use that syntax. It was just my suggestion. I have no problem if you prefer to use "standard" syntax, as long as the misleading braces are removed. Other acceptable solutions for me are:
```csharp
if (Receiver == null) return;
Receiver.SteamBoosterChangeTo(ToState, Target);
```
or
```csharp
if (Receiver != null)
{
    Receiver.SteamBoosterChangeTo(ToState, Target);
}
```
or
```csharp
if (Receiver != null) Receiver.SteamBoosterChangeTo(ToState, Target);
```

Comment (peternewell) [reply Source/Orts.Simulation/Common/Commands.cs]: Ok, I have adopted one of your suggested solutions.

Comment (cesarBLG) [reply Source/Orts.Simulation/Common/Commands.cs]: I'm sorry for being so nitpicking, but the indentation of `Receiver.SteamBoosterChangeTo(ToState, Target);` is still incorrect. I think it is important that we follow the coding guidelines to make code more readable for others. I could have corrected it myself, but you disabled "allow edits from maintainers".

Comment (peternewell) [reply Source/Orts.Simulation/Common/Commands.cs]: I have moved it across now. Thanks for checking.

**Review by cesarBLG — APPROVED** [review]
Thanks, and sorry for the inconvenience of the multiple style suggestions.

**Comment by peternewell** [comment]
Thanks Cesar.

**Pull request merged**
